### DISCUSSION TITLE
Problems deploying a Grails 5.2.x war app on Tomcat 10.1.x

### DISCUSSION THREAD

**Discussion by preethitp** [apache/grails-core#13600] [General]
Issue Description

I am currently working with the following setup:

Operating System: RHEL 8
Java Version: JDK 17.0.12
Grails Version: 5.2.2
Spring Framework Version: 6.1.12
Spring Boot Version: 3.3.3
Groovy Version: 3.x
Tomcat Version: 10.1.28 (64-bit tar.gz)
IDE: IntelliJ 2021.1.7
Gradle Version: 7.5
I am creating a WAR package for my application and deploying it to the webapps folder on Tomcat 10.1.28. However, I encountered an issue during deployment. Specifically, I am receiving the following error message:

`Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [grails.boot.config.GrailsApplicationPostProcessor]: Factory method 'grailsApplicationPostProcessor' threw exception with message: org/springframework/core/NestedIOException`

Could you please advise if this issue might be related to a version mismatch between the libraries or frameworks I am using?

Additionally, do you have any information on the expected release date for Grails 7.x?

Thank you for your assistance!

**Comment by jeffscottbrown** [comment]
Is Tomcat 10 supported with Grails 5?

**Reply by preethitp** [reply]
Tomcat 10 supports the Servlet 5.0 specification. Grails 5 is built on Spring Boot 3.x, which is compatible with Servlet 5.0. Therefore, Grails 5 should work with Tomcat 10 in terms of servlet API compatibility.

**Reply by jamesfredley** [reply]
Grails 5 and Grails 6 both shipped with Tomcat 9.0.x based on the Spring Boot dependencies:  

Grails 5 with Spring Boot 2.7.5 https://docs.spring.io/spring-boot/docs/2.7.5/reference/html/dependency-versions.html

Grails 6 with Spring Boot 2.7.18 https://docs.spring.io/spring-boot/docs/2.7.18/reference/html/dependency-versions.html

Grails 7 will support Spring Boot 3.2.x or 3.3.x  with Tomcat 10.1.x https://docs.spring.io/spring-boot/docs/3.2.9/reference/html/dependency-versions.html

You can view Grails 7 progress on https://github.com/orgs/grails/projects/3

**Reply by preethitp** [reply]
Thank you for the reply.

Is there any tentative release date for grails 7?

**Reply by jeffscottbrown** [reply]
> Is there any tentative release date for grails 7?

There is not at this point.  It will likely be Q4 of this year or Q1 of next, but the team is still making progress on the release plan.
